src/dockerconfig/docker_config.go

```diff
@@ -25,47 +25,30 @@ type DockerAuth struct {
 	Password string `json:"password"`
 }
 
-func NewDockerConfig(ctx context.Context, apiReader client.Reader, dynakube dynatracev1beta1.DynaKube) (*DockerConfig, error) {
-	var pullSecret corev1.Secret
-	if err := apiReader.Get(ctx, client.ObjectKey{Name: dynakube.PullSecret(), Namespace: dynakube.Namespace}, &pullSecret); err != nil {
-		log.Info("failed to load pull secret", "dynakube", dynakube.Name)
-		return nil, err
-	}
-	dockerAuths, err := parseDockerAuthsFromSecret(&pullSecret)
-	if err != nil {
-		log.Info("failed to parse pull secret content", "dynakube", dynakube.Name)
-		return nil, err
-	}
-
+func NewDockerConfig(apiReader client.Reader, dynakube dynatracev1beta1.DynaKube) *DockerConfig {
 	dockerConfig := DockerConfig{
 		ApiReader:     apiReader,
+		Auths:         make(map[string]DockerAuth),
 		Dynakube:      &dynakube,
-		Auths:         dockerAuths,
 		SkipCertCheck: dynakube.Spec.SkipCertCheck,
```

Comment:
I dont think we would need SkipCertCheck to be part of the struct, as we have the dynakube as a whole anyway

Reply:
made it a method, so we don't have to dig for it 👍